packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -311,7 +311,7 @@ class PouchLink extends CozyLink {
   }
 
   async ensureIndex(doctype, query) {
-    const fields = getIndexFields(query)
+    const fields = query.indexedFields || getIndexFields(query)
```

Comment:
Can you add a test ? It'd be a little easier to understand what was not working before.